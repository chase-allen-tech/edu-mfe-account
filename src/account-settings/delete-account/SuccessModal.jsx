import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
import { Modal } from '@edx/paragon';

import messages from './messages';

export const SuccessModal = (props) => {
  const { status, intl, onClose } = props;
  return (
    <Modal
      open={status === 'deleted'}
      title={intl.formatMessage(messages['account.settings.delete.account.modal.after.header'])}
      body={(
        <div>
          <p className="h6">
            {intl.formatMessage(messages['account.settings.delete.account.modal.after.text'])}
          </p>
        </div>
      )}
      closeText={intl.formatMessage(messages['account.settings.delete.account.modal.after.button'])}
      renderHeaderCloseButton={false}
      onClose={onClose}
    />
  );
};

SuccessModal.propTypes = {
  status: PropTypes.oneOf(['confirming', 'pending', 'deleted', 'failed']),
  intl: intlShape.isRequired,
  onClose: PropTypes.func.isRequired,
};

SuccessModal.defaultProps = {
  status: null,
};

export default injectIntl(SuccessModal);
