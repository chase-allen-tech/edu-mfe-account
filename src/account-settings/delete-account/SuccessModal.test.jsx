import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { IntlProvider, injectIntl } from '@edx/frontend-platform/i18n';

// Modal creates a portal.  Overriding ReactDOM.createPortal allows portals to be tested in jest.
ReactDOM.createPortal = node => node;

import { SuccessModal } from './SuccessModal'; // eslint-disable-line import/first

const IntlSuccessModal = injectIntl(SuccessModal);

describe('SuccessModal', () => {
  let props = {};

  beforeEach(() => {
    props = {
      onClose: jest.fn(),
      status: null,
    };
  });

  it('should match default closed success modal snapshot', () => {
    let tree = renderer.create((
      <IntlProvider locale="en"><IntlSuccessModal {...props} /></IntlProvider>))
      .toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create((
      <IntlProvider locale="en"><IntlSuccessModal {...props} status="confirming" /></IntlProvider>))
      .toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create((
      <IntlProvider locale="en"><IntlSuccessModal {...props} status="pending" /></IntlProvider>))
      .toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create((
      <IntlProvider locale="en"><IntlSuccessModal {...props} status="failed" /></IntlProvider>))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match open success modal snapshot', () => {
    const tree = renderer
      .create((
        <IntlProvider locale="en">
          <IntlSuccessModal
            {...props}
            status="deleted" // This will cause 'modal-backdrop' and 'show' to appear on the modal as CSS classes.
          />
        </IntlProvider>
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
