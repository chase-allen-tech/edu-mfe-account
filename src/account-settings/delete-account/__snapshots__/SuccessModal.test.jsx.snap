// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessModal should match default closed success modal snapshot 1`] = `
Array [
  <div
    className="fade"
    role="presentation"
  />,
  <div
    className="modal fade"
    role="presentation"
  >
    <div
      aria-labelledby="id1"
      aria-modal={true}
      className=""
      role="dialog"
    >
      <div
        data-focus-guard={true}
        style={
          Object {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
      <div
        data-focus-guard={true}
        style={
          Object {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
      <div
        data-focus-lock-disabled="disabled"
        onBlur={[Function]}
        onFocus={[Function]}
        onScrollCapture={[Function]}
        onTouchMoveCapture={[Function]}
        onWheelCapture={[Function]}
      >
        <div
          className="modal-content"
        >
          <div
            className="modal-header"
          >
            <h2
              className="modal-title"
              id="id1"
            >
              We're sorry to see you go!  Your account will be deleted shortly.
            </h2>
          </div>
          <div
            className="modal-body"
          >
            <div>
              <p
                className="h6"
              >
                Account deletion, including removal from email lists, may take a few weeks to fully process through our system. If you want to opt-out of emails before then, please unsubscribe from the footer of any email.
              </p>
            </div>
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-link"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
      <div
        data-focus-guard={true}
        style={
          Object {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
    </div>
  </div>,
]
`;

exports[`SuccessModal should match default closed success modal snapshot 2`] = `
Array [
  <div
    className="fade"
    role="presentation"
  />,
  <div
    className="modal fade"
    role="presentation"
  >
    <div
      aria-labelledby="id2"
      aria-modal={true}
      className=""
      role="dialog"
    >
      <div
        data-focus-guard={true}
        style={
          Object {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
      <div
        data-focus-guard={true}
        style={
          Object {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
      <div
        data-focus-lock-disabled="disabled"
        onBlur={[Function]}
        onFocus={[Function]}
        onScrollCapture={[Function]}
        onTouchMoveCapture={[Function]}
        onWheelCapture={[Function]}
      >
        <div
          className="modal-content"
        >
          <div
            className="modal-header"
          >
            <h2
              className="modal-title"
              id="id2"
            >
              We're sorry to see you go!  Your account will be deleted shortly.
            </h2>
          </div>
          <div
            className="modal-body"
          >
            <div>
              <p
                className="h6"
              >
                Account deletion, including removal from email lists, may take a few weeks to fully process through our system. If you want to opt-out of emails before then, please unsubscribe from the footer of any email.
              </p>
            </div>
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-link"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
      <div
        data-focus-guard={true}
        style={
          Object {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
    </div>
  </div>,
]
`;

exports[`SuccessModal should match default closed success modal snapshot 3`] = `
Array [
  <div
    className="fade"
    role="presentation"
  />,
  <div
    className="modal fade"
    role="presentation"
  >
    <div
      aria-labelledby="id3"
      aria-modal={true}
      className=""
      role="dialog"
    >
      <div
        data-focus-guard={true}
        style={
          Object {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
      <div
        data-focus-guard={true}
        style={
          Object {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
      <div
        data-focus-lock-disabled="disabled"
        onBlur={[Function]}
        onFocus={[Function]}
        onScrollCapture={[Function]}
        onTouchMoveCapture={[Function]}
        onWheelCapture={[Function]}
      >
        <div
          className="modal-content"
        >
          <div
            className="modal-header"
          >
            <h2
              className="modal-title"
              id="id3"
            >
              We're sorry to see you go!  Your account will be deleted shortly.
            </h2>
          </div>
          <div
            className="modal-body"
          >
            <div>
              <p
                className="h6"
              >
                Account deletion, including removal from email lists, may take a few weeks to fully process through our system. If you want to opt-out of emails before then, please unsubscribe from the footer of any email.
              </p>
            </div>
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-link"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
      <div
        data-focus-guard={true}
        style={
          Object {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
    </div>
  </div>,
]
`;

exports[`SuccessModal should match default closed success modal snapshot 4`] = `
Array [
  <div
    className="fade"
    role="presentation"
  />,
  <div
    className="modal fade"
    role="presentation"
  >
    <div
      aria-labelledby="id4"
      aria-modal={true}
      className=""
      role="dialog"
    >
      <div
        data-focus-guard={true}
        style={
          Object {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
      <div
        data-focus-guard={true}
        style={
          Object {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
      <div
        data-focus-lock-disabled="disabled"
        onBlur={[Function]}
        onFocus={[Function]}
        onScrollCapture={[Function]}
        onTouchMoveCapture={[Function]}
        onWheelCapture={[Function]}
      >
        <div
          className="modal-content"
        >
          <div
            className="modal-header"
          >
            <h2
              className="modal-title"
              id="id4"
            >
              We're sorry to see you go!  Your account will be deleted shortly.
            </h2>
          </div>
          <div
            className="modal-body"
          >
            <div>
              <p
                className="h6"
              >
                Account deletion, including removal from email lists, may take a few weeks to fully process through our system. If you want to opt-out of emails before then, please unsubscribe from the footer of any email.
              </p>
            </div>
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-link"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
      <div
        data-focus-guard={true}
        style={
          Object {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={-1}
      />
    </div>
  </div>,
]
`;

exports[`SuccessModal should match open success modal snapshot 1`] = `
Array [
  <div
    className="modal-backdrop show"
    role="presentation"
  />,
  <div
    className="modal show d-block"
    role="presentation"
  >
    <div
      aria-labelledby="id5"
      aria-modal={true}
      className="modal-dialog"
      role="dialog"
    >
      <div
        data-focus-guard={true}
        style={
          Object {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={0}
      />
      <div
        data-focus-guard={true}
        style={
          Object {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={1}
      />
      <div
        data-focus-lock-disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onScrollCapture={[Function]}
        onTouchMoveCapture={[Function]}
        onTouchStart={[Function]}
        onWheelCapture={[Function]}
      >
        <div
          className="modal-content"
        >
          <div
            className="modal-header"
          >
            <h2
              className="modal-title"
              id="id5"
            >
              We're sorry to see you go!  Your account will be deleted shortly.
            </h2>
          </div>
          <div
            className="modal-body"
          >
            <div>
              <p
                className="h6"
              >
                Account deletion, including removal from email lists, may take a few weeks to fully process through our system. If you want to opt-out of emails before then, please unsubscribe from the footer of any email.
              </p>
            </div>
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-link"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
      <div
        data-focus-guard={true}
        style={
          Object {
            "height": "0px",
            "left": "1px",
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "top": "1px",
            "width": "1px",
          }
        }
        tabIndex={0}
      />
    </div>
  </div>,
]
`;
